src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -104,7 +110,15 @@ public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
                             {
                                 if (listPackageArgs.ReportType != ReportType.Default)  // generic list package is offline -- no server lookups
                                 {
-                                    await GetRegistrationMetadataAsync(packages, listPackageArgs);
+                                    IEnumerable<Lazy<INuGetResourceProvider>> providers = Repository.Provider.GetCoreV3();
+                                    IEnumerable<PackageSource> sources = listPackageArgs.PackageSources;
+                                    foreach (PackageSource source in sources)
+                                    {
+                                        var sourceRepository = Repository.CreateSource(providers, source, FeedType.Undefined);
+                                        _sourceRepositoryCache[(providers, source)] = sourceRepository;
+                                    }
```

Comment:
Consider putting this new code in a method, so that the code block is still easy to read:

```cs
if (condition)
{
  DoThing1();
  DoThing2();
  DoThing3();
}
```

Reply:
@zivkan 
I addressed it. Please review again.